### PR TITLE
fix #271198: Custom noteheads implementation

### PR DESCRIPTION
**Pull request by anatoly-os**
see also https://github.com/musescore/MuseScore/pull/3620

### CODE REVIEW

**Comment by anatoly-os** [comment]
@lasconic could you please review the UX of editdrumset widget? Important question is, why do we need and whether it is possible to remove "invalid" item from notehead groups combobox? Can I add "Custom" item and what additional work to keep compatibility should be done?

**Comment by lasconic** [comment]
Based on https://github.com/musescore/MuseScore/pull/3620

**Comment by anatoly-os** [comment]
Squashed commits. The PR is ready for merging.

File: mscore/editdrumset.cpp [hunk 1]
```diff
@@ -239,12 +307,25 @@ void EditDrumset::apply()
 //   itemChanged
```
Comment (lasconic) [inline mscore/editdrumset.cpp]: Can you fix the comments "itemChanged" and put it before the right method ? Also add comment for new methods please.

**Comment by lasconic** [comment]
*why do we need and whether it is possible to remove "invalid" item from notehead groups combobox?*

I can't find a way to remove a "pitch" from the drumset without the invalid item. 

* Select an unassigned pitch, change to normal, press OK, see the drum palette has a new entry 
* You want to delete this entry. You can't.

**Pull request merged**